runtime/src/lib.rs

```diff
@@ -136,18 +136,36 @@ pub trait Spawner: Clone + Send + Sync + 'static {
         F: Future<Output = T> + Send + 'static,
         T: Send + 'static;
 
-    /// Enqueue a blocking task to be executed.
+    /// Enqueue a blocking task on a shared thread.
     ///
-    /// This method is designed for synchronous, potentially long-running operations that should
-    /// not block the asynchronous event loop. The task starts executing immediately, and the
-    /// returned handle can be awaited to retrieve the result.
+    /// This method runs synchronous operations in a shared thread pool. Tasks spawned using this method
+    /// should eventually finish on their own (to avoid blocking the thread pool).
+    ///
+    /// The task starts executing immediately, and the returned handle can be awaited to retrieve the
+    /// result.
     ///
     /// # Warning
     ///
     /// Blocking tasks cannot be aborted.
     fn spawn_blocking<F, T>(self, f: F) -> Handle<T>
     where
-        F: FnOnce() -> T + Send + 'static,
+        F: FnOnce(Self) -> T + Send + 'static,
```

Comment:
Passed through `Context` here in case it is useful (not sure why I didn't do this to begin with, as everything we have is already compatible with a multi-threaded env).